joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -277,12 +278,13 @@ controller_interface::return_type JointTrajectoryController::update(
         if (use_closed_loop_pid_adapter_)
         {
           // Update PIDs
-          for (auto i = 0ul; i < dof_; ++i)
+          for (auto i = 0ul; i < num_cmd_joints_; ++i)
           {
-            tmp_command_[i] = (state_desired_.velocities[i] * ff_velocity_scale_[i]) +
-                              pids_[i]->computeCommand(
-                                state_error_.positions[i], state_error_.velocities[i],
-                                (uint64_t)period.nanoseconds());
+            size_t index = map_cmd_to_joints_[i];
+            tmp_command_[index] = (state_desired_.velocities[index] * ff_velocity_scale_[i]) +
```

Comment:
no you are right, this is a bug and should be [index] (I should name it better)

This is a general problem because I don't want to change the parameter file to break current setups. the gains-structure would be only necessary for command_joints, but this parameter is optional -> the generated map by the GPL has to be refreshed if it is copied over from joints-parameter. (should work)

but the angle_wraparound parameter will be necessary for all joints. It does not fit well into the gains-structure, maybe I should change this with a deprecation warning when using the old structure.

Reply:
Let me think about it a bit. I will let you know